### ISSUE TITLE
multiFormatSave broken as GlobalScope.DialogLibraries.isLibraryLoaded(EXTENSION_NAME) not working

### ISSUE THREAD

**Issue by ccheng** [ertaiNL/multiFormatSave#9]
Not sure when this became an issue as it used to work.  At some point LibreOffice must've updated to version 6.3.4.2 on Lubuntu 19.04 while I had MultiFormatSave 1.5.6 installed.  Then it stopped working:

The error occurs on line 78 (or 76 if viewed in LO macro editor) in `multiFormatSave.xba`.  Specifically `GlobalScope.DialogLibraries.isLibraryLoaded(EXTENSION_NAME)` throws a `NoSuchElementException.`

This problem occurs despite removing and re-adding the extension, with re-opening LO between each step.

I did figure out a work-around: open `~/.config/libreoffice/4/user/basic/dialog.xlc` and ensure it's contents is:
```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE library:libraries PUBLIC "-//OpenOffice.org//DTD OfficeDocument 1.0//EN" "libraries.dtd">
<library:libraries xmlns:library="http://openoffice.org/2000/library" xmlns:xlink="http://www.w3.org/1999/xlink">
 <library:library library:name="multiFormatSave" xlink:href="$(USER)/uno_packages/cache/uno_packages/lu23853otd9xi.tmp_/multiFormatSave_v1-5-6.oxt/multiFormatSave/dialog.xlb/" xlink:type="simple" library:link="false"/>
</library:libraries>
```

The `lu23853otd9xi.tmp_` must match where LO installed the multiFormatSave extension, of course.

Unfortunately, I'm not sure if that file existed before my bug hunting, or whether my debugging created that file since I created a Dialog in the LO macro editor as part of the bug hunt.

In any case, the key line `xlink:href="$(USER)/uno_packages/cache/uno_packages/lu23853otd9xi.tmp_/multiFormatSave_v1-5-6.oxt/multiFormatSave/dialog.xlb/"` seems to be the "active ingredient", letting LO know where to find the proper `dialog.xlb` file as it could've find it before.

Any ideas on how this problem occurred, and whether it could be fixed?  Thanks!

**Comment by ertaiNL** [comment]
HI,

I am using 6.3.5.2 on Ubuntu LTS 18.84.4 and it works for me. I don't really use the plugin myself anymore. But more people are getting issues with it. So I am not sure what goes wrong.

My own dialog.xlc contains: 

`<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE library:libraries PUBLIC "-//OpenOffice.org//DTD OfficeDocument 1.0//EN" "libraries.dtd">
<library:libraries xmlns:library="http://openoffice.org/2000/library" xmlns:xlink="http://www.w3.org/1999/xlink">
 <library:library library:name="Standard" xlink:href="$(USER)/basic/Standard/dialog.xlb/" xlink:type="simple" library:link="false"/>
</library:libraries>`

If you insert that in yours. Does the plugin work then?

**Comment by ccheng** [comment]
Hi!
It's strange.  I didn't get around to testing it until now, but it started working without the workaround I wrote up before.

In that time there's been upgrades, so I'm now using LibreOffice version 6.4.2.2 on Lubuntu 19.10 and MultiFormatSave 1.5.6.

I've been using it with the dialog.xlc workaround I wrote up before, and it works.

I tried the dialog.xlc you suggested, and it worked.

I deleted dialog.xlc and reinstalled the extension, and it *still* worked!

As a sanity check, I modified dialog.xlc with a wrong xlink:href path (so it doesn't point to the extension), and it finally stopped working.

On my Mac, the dialog.xlc contains what your dialog.xlc contains.  So it's probably best to have a dialog.xlc that has what yours contains.

I guess this issue is resolved!

PS:  I do use this extension all the time, so Thanks for spending time on this!
